File: PassangerApp/lib/Models/User.model.dart
class User {
  String? name;
  String? email;
  String? image;
  String? password;
  String? mobileNumber;
  String? role;

  User({
    this.name,
    this.email,
    this.image,
    this.password,
    this.mobileNumber,
    this.role,
  });

  User.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    email = json['email'];
    image = json['image'];
    password = json['password'];
    mobileNumber = json['mobileNumber'];
    role = json['role'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this.name;
    data['email'] = this.email;
    data['image'] = this.image;
    data['password'] = this.password;
    data['mobileNumber'] = this.mobileNumber;
    data['role'] = this.role;
    return data;
  }
}

File: Backend/models/User.model.js
// make model for user 

const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const userSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    address: {
        type: String,

    },
    image: {
        type: String, default: '',
    },

    password: {
        type: String,
        required: true
    },
    location: {
        type: {
            type: String,
            enum: ['Point'],
            default: 'Point'
        },
        coordinates: {
            type: [Number],
            index: '2dsphere',
            default: [0, 0]
        }
    },
    mobileNumber: {
        type: String,
        required: true
    },
    vehicle: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Vehicle',
        default: null
    },
    idCard: {
        type: String,
        default: ''
    },
    drivingLivence: {
        type: String,
        default: ''

    },
    vehicleRegistration: {
        type: String,
        default: ''
    },
    role: {
        type: String
    },
    rating: {
        type: Number,
        default: 0
    },

    status: {
        type: String,
        default: 'active'
    },

    createdAt: {
        type: Date,
        default: Date.now
    },
});

// userSchema.index({ location: "2dsphere" });

const User = mongoose.model('User', userSchema);

module.exports = User;
